src/modules/deliveryoptimization/src/lib/DeliveryOptimization.c

```diff
@@ -0,0 +1,417 @@
+                    {
+                        const char* cacheHost = json_value_get_string(currentValue);
+                        json_object_set_string(newObject, g_cacheHostConfigName, cacheHost);
+                    }
+                    else if ((0 == strcmp(name, g_desiredCacheHostSourceSettingName)) && (JSONNumber == json_value_get_type(currentValue)))
+                    {
+                        const int cacheHostSource = (int)json_value_get_number(currentValue);
+                        if ((cacheHostSource >= 0) && (cacheHostSource <= 3))
+                        {
+                            json_object_set_number(newObject, g_cacheHostSourceConfigName, cacheHostSource);
+                        }
+                        else 
+                        {
+                            OsConfigLogError(DeliveryOptimizationGetLog(), "MmiSet called with invalid cacheHostSource (%d)", cacheHostSource);
+                            status = EINVAL;
+                        }
+                    }
+                    else if ((0 == strcmp(name, g_desiredCacheHostFallbackSettingName)) && (JSONNumber == json_value_get_type(currentValue)))
+                    {
+                        const int cacheHostFallback = (int)json_value_get_number(currentValue);
+                        json_object_set_number(newObject, g_cacheHostFallbackConfigName, cacheHostFallback);
+                    }
+                    else if ((0 == strcmp(name, g_desiredPercentageDownloadThrottleSettingName)) && (JSONNumber == json_value_get_type(currentValue)))
+                    {
+                        const int percentageDownloadThrottle = (int)json_value_get_number(currentValue);
+                        if ((percentageDownloadThrottle >= 0) && (percentageDownloadThrottle <= 100))
+                        {
+                            json_object_set_number(newObject, g_percentageDownloadThrottleConfigName, percentageDownloadThrottle);
+                        }
+                        else 
+                        {
+                            OsConfigLogError(DeliveryOptimizationGetLog(), "MmiSet called with invalid percentageDownloadThrottle (%d)", percentageDownloadThrottle);
+                            status = EINVAL;
+                        }
+                    }
+                }
+
+                if (JSONSuccess != json_serialize_to_file_pretty(newValue, g_deliveryOptimizationConfigFile))
+                {
+                    OsConfigLogError(DeliveryOptimizationGetLog(), "MmiSet failed to write JSON file (%s)", g_deliveryOptimizationConfigFile);
+                    status = EIO;
+                }
+
+                json_value_free(newValue);
+            }
+            else
+            {
+                OsConfigLogError(DeliveryOptimizationGetLog(), "MmiSet failed to parse JSON (%.*s)", payloadSizeBytes, payload);

```

Comment:
>(%.*s) [](http://example.com/codeflow?start=92&length=6)

() are used as just that, parenthesis, in similar traces. Not as string separators. If you want to log that JSON text clearly marked, use '', like this:

"MmiSet failed to parse JSON '%.*s"

This comment applies to all other instances in this PR.

Reply:
Fixed, thanks!